integration-tests/vertx-http-compressors/some/src/test/java/io/quarkus/compressors/it/RESTEndpointsTest.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.compressors.it;
+
+import static io.quarkus.compressors.it.Testflow.runTest;
+
+import java.net.URL;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import io.quarkus.test.common.http.TestHTTPResource;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class RESTEndpointsTest {
+
+    @TestHTTPResource(value = "/compressed")
+    URL url;
+
+    @ParameterizedTest
+    @CsvSource(value = {
+    //@formatter:off
+    // Context | Accept-Encoding | Content-Encoding | Content-Length
+    "/yes/text | deflate,gzip,br | gzip             | 2414",
+    "/yes/text | deflate         | deflate          | 2402",
+    "/no/text  | deflate,gzip,br | null             | 6483",
+    "/yes/json | deflate         | deflate          | 2402",
+    "/no/json  | deflate,gzip,br | null             | 6483",
+    "/yes/xml  | deflate,gzip,br | gzip             | 2414",
+    "/no/xml   | deflate,gzip,br | null             | 6483",
```

Comment:
Can't we test a marker in the binary rather than the content-length? This is going to be very brittle if some changes are made to any of the compression impl.

Reply:
The native part is built in the brotli4 repo, It's not imported. I had some `printf` in the C part there, so I know for sure it really loads what I think it does.

Reply:
@gsmet It kept nagging at me, so I conducted a quick experiment focused not on the compressor as such, but on the *default* parameters that are indeed changed in Netty.

```
Vert.x: 4.5.8, Brotli: 1.16.0, Netty: 4.1.110.Final
Uncompressed
content-length: 80948

content-encoding: gzip
content-length: 3530

content-encoding: deflate
content-length: 3518

content-encoding: br
content-length: 2758

Vert.x: 4.4.9, Brotli: 1.16.0, Netty: 4.1.108.Final
Uncompressed
content-length: 80948

content-encoding: gzip
content-length: 3530

content-encoding: deflate
content-length: 3518

content-encoding: br
content-length: 2758

Vert.x: 4.3.8, Brotli: 1.16.0, Netty: 4.1.87.Final
Uncompressed
content-length: 80948

content-encoding: gzip
content-length: 3530

content-encoding: deflate
content-length: 3518

content-encoding: br
content-length: 2757
```

Note that the  Netty: 4.1.87.Final  with Brotli compressed my test text to 2757 bytes whereas newer Netty to 2758.
I will add a tolerance window.

Reply:
Again, can't we check a marker in the content instead? That's what I originally asked for and I don't want to be in the situation we already encountered with AWT and other things, where we have to update numeric values when a new implementation of Brotli appears.

Reply:
https://github.com/quarkusio/quarkus/pull/40750#issuecomment-2148046690